server.go

```diff
@@ -207,16 +221,31 @@ func (s *server) UpdatePin(ctx context.Context, request v0.UpdatePinRequestObjec
 	}
 
 	log.Println("found smartlock auth", *authRes.Payload[index].ID, "claim name", c.Email, "auth ID", *user.AccountUserID, "auth name", *authRes.Payload[index].Name)
+	log.Println("allowed week days", s.cfg.Nuki.AllowedWeekDays)
+	body := &models.SmartlockAuthUpdate{
+		Code:          int32(request.Body.Code),
+		Enabled:       true,
+		AccountUserID: *user.AccountUserID,
+		AllowedWeekDays:     ptr(int32(s.cfg.Nuki.AllowedWeekDays)),
+		}
+
+	if s.cfg.Nuki.AllowedFromDate != nil && s.cfg.Nuki.AllowedUntilDate != nil{
```

Comment:
The annoying thing is that the Body is expecting non-pointer values for not required fields. So at the moment I don't see a way to avoid the if-statement. 

An idea would be to not use pointers in the config and always apply the default values. I think technically this should make no difference. 

ref: https://github.com/go-swagger/go-swagger/issues/1188

Reply:
I tried with the default value, Nuki API wasn't happy (put some date in 2001...)